pkg/providers/vsphere/virtualmachine/snapshot.go

```diff
@@ -259,18 +257,61 @@ func GetSnapshotSize(logger logr.Logger,
 		if len(childMostFileKeys) > 0 {
 			logger.V(5).Info("Adding file key for the child most delta disk of the snapshot",
 				"fileKey", childMostFileKeys)
+			var curDiskTotal int64
 			for _, fileKey := range childMostFileKeys {
-				total += fileKeyMap[fileKey]
+				curDiskTotal += fileKeyMap[fileKey]
+			}
+
+			if fcdDeviceKeySet.Has(disk.Key) {
+				// A VM snapshot creates a delta disk for all disks of a VM --
+				// including PVCs that are backed by FCDs. Since the usage of
+				// the delta disks created on PVCs will already be
+				// reported by the VolumeSnapshot SPU, we skip adding those to
+				// the total size, which will be used by calculating the usage of
+				// the VirtualMachineSnapshot SPU.
+				// But we return the FCD's total separately for other usage.
+				logger.V(5).Info("Current disk is FCD", "diskKey", disk.Key)
+				fcdTotal += curDiskTotal
+			} else {
+				total += curDiskTotal
 			}
 		}
 	}
 
-	return total, nil
+	return total, fcdTotal, nil
+}
+
+// GetAllSnapshotSize calculates the size of a all snapshots of the VM in bytes.
+// It returns two sizes, first one includes the memory file, the vmdk files,
+// and the vmsn file, excludes FCDs. Second one only includes FCDs.
+func GetAllSnapshotSize(
+	logger logr.Logger,
+	moVM mo.VirtualMachine) (int64, int64, error) {
+	vmLayout := moVM.LayoutEx
```

Comment:
There's no reason to alias `moVM.LayoutEx`. And if you did, always keep a blank line between the function signature and the first line in it.